src/exporters/mod.rs

```diff
@@ -595,7 +597,15 @@ impl MetricGenerator {
     fn gen_kubernetes_pods_basic_metadata(&mut self) {
         if self.watch_kubernetes {
             if let Some(kubernetes) = self.kubernetes_client.as_mut() {
-                if let Ok(pods_result) = kubernetes.list_pods("".to_string()) {
+                let node_name = env::var("KUBERNETES_NODE_NAME").unwrap_or_default();
+                let selector = format!("spec.nodeName={}", node_name);
+                let mut list_options = ListOptional {
+                    ..Default::default()
+                };
+                if !node_name.is_empty() {
+                    list_options.field_selector = Some(&selector);
```

Comment:
This was the cleanest way I could see to set the field selector as it uses a `&str` but it's very possible I'm missing something :)